### ISSUE TITLE
Mutating snapshot

### ISSUE THREAD

**Issue by 0xedb** [pmndrs/eslint-plugin-valtio#13]
> Snapshots are made to be used with react, so we recommend them for mutating, and not states directly.

```ts
const state = proxy({ count: 0})

function App() {
  const handleClick = () => {
    state.count = state.count + 1 // Mutating a proxy object itself. this might not be expected as it's not reactive.
  }
 return <button onClick={handleClick}>mutate</button> 
}
```

What's the recommended way of mutating a snapshot? Thanks.

**Comment by dai-shi** [comment]
Oh, your code seems perfectly fine. So, it might be a false positive with the rule?
@Aslemammad ? cc @barelyhuman

**Comment by riccardoscalco** [comment]
The rule seems to contradict the rule of thumb suggested in the [readme](https://github.com/pmndrs/valtio#react-via-usesnapshot):

> Rule of thumb: read from snapshots, mutate the source.

**Comment by dai-shi** [comment]
@riccardoscalco Yeah, let's revisit it. How about this?
"Rule of thumb: use snapshots to render, otherwise use the source."
Does it make sense???

**Comment by riccardoscalco** [comment]
@dai-shi I am a bit confuse, I cannot say if it makes sense.
Could you provide some example for the rule:

> Snapshots are made to be used with react, so we recommend them for mutating, and not states directly.

Thanks.

**Comment by dai-shi** [comment]
That's an error in readme. Let's fix it. Sorry, I should have noticed in advance.

- Snapshots are not mutable by concept and they are frozen, so whenever you want to mutate, mutate original proxies.
- Snapshots are reactive source, use them in render (= to generate JSX).
